app/src/main/java/ar/edu/utn/frba/mobile/a2019c1/mercury/NotificationService.kt

```diff
@@ -0,0 +1,130 @@
+            notificationChannel.enableVibration(true)
+            notificationChannel.setShowBadge(true)
+            notificationChannel.enableLights(true)
+            notificationChannel.lightColor = Color.parseColor("#e8334a")
+            notificationChannel.description = "Para el envio de itinerarios"//Agregar a strings.xml
+            notificationChannel.lockscreenVisibility = Notification.VISIBILITY_PUBLIC
+            notificationManager.createNotificationChannel(notificationChannel)
+        }
+
+    }
+
+    companion object {
+
+        const val CHANNEL_ID = "ar.edu.utn.frba.mobile.a2019c1.mercury.Itinerarios"
+        const val CHANNEL_NAME = "Itinerarios"
+    }
+
+
+    override fun onHandleIntent(intent: Intent?) {
+
+        //Create Channel
+        createChannel()
+
+
+        var timestamp: Long = 0
+        var title = "Sample Notification"
+        var message = "You have received a sample notification. This notification will take you to the details page."
+
+        if (intent != null && intent.extras != null) {
+            timestamp = intent.extras!!.getLong("timestamp")
+            title = intent.extras!!.getString("title",title)
+            message = intent!!.extras!!.getString("message",message)
+        }
+
+
+        if (timestamp > 0) {
+
+
+            val context = this.applicationContext
+            var notificationManager: NotificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+            val notifyIntent = Intent(this, MainActivity::class.java) //Se elige la activity que se va a abrir
+
+
+
+            notifyIntent.putExtra("title", title)
+            notifyIntent.putExtra("message", message)
+            notifyIntent.putExtra("notification", true)
+
+            notifyIntent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
+
+            val calendar = Calendar.getInstance()
+            calendar.timeInMillis = timestamp
+
+
+            val pendingIntent = PendingIntent.getActivity(context, 0, notifyIntent, PendingIntent.FLAG_UPDATE_CURRENT)
+            val res = this.resources
+            val uri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION)
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+
+
+                mNotification = Notification.Builder(applicationContext, CHANNEL_ID)
+                    // Set the intent that will fire when the user taps the notification
+                    .setContentIntent(pendingIntent)
+                    .setSmallIcon(R.drawable.ic_app_icon)
+                    .setLargeIcon(BitmapFactory.decodeResource(res, R.mipmap.ic_launcher))
+                    .setAutoCancel(true)
+                    .setContentTitle(title)
+                    .setStyle(Notification.BigTextStyle()
+                        .bigText(message))
+                    .setContentText(message).build()
+            } else {
+
+                mNotification = Notification.Builder(this)
```

Comment:
Podes hacer un metodo que reciba `this `o `applicationContext, CHANNEL_ID`, que haga todos los seteos comunes y despues cada uno haga los propios asi no se repite tanto

Reply:
Si, pero creo que voy a borrar el else mejor, cambia un par de cosas dependiendo del SO, pero importa Oreo en adelante nomas